src/pages/PremiumVideoCall/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx

```diff
@@ -93,12 +92,20 @@ export default function DeviceSelectionScreen({
 
     const handleJoin = async () => {
         setIsLoading(true)
-        const api = new Api(token)
-        const { jwtAccessToken } = await api.joinRoom(roomId, 'test-password') // TODO pass the password variable
-        dispatch(actions.setRoomId(roomId))
-        dispatch(actions.setHostName(name))
-        connect(jwtAccessToken, roomId)
-        setIsLoading(false)
+        try {
+            const twilioVideoToken = await getTwilioVideoToken(
```

Comment:
this could probably be within an action, and dispatch the roomid and hostName within the action.

Reply:
How would you call the function to make that clear that it
- gets a video token
- dispatches the hostName
- dispatches the roomName

Reply:
initVideoToken
loadTwilioRoomInfo

we don't need to know what will dispatch the action, just what it's doing overall.

Reply:
I think a nice approach could be to create a `connectToRoom` async action, that is dispatched once clicking on `Join`. This action would:
- dispatch the `hostName` & `roomId`
- take care of showing/hiding the loader

We'd thus need to move the loader indicator to redux. I'd treat this in a separate low priority PR although (cleaner code)

Reply:
ok